overlord/ifacestate/ifacemgr.go

```diff
@@ -275,9 +275,23 @@ func (m *InterfaceManager) doRemoveSnapSecurity(task *state.Task, _ *tomb.Tomb)
 	if err != nil {
 		return err
 	}
-	// TODO: remove all connections from the state
-	if err := m.repo.RemoveSnap(snapName); err != nil {
 
+	conns, err := getConns(task.State())
```

Comment:
Almost.. snap remove may be used to remove one of _several_ snaps. We cannot discard connections if we still have a snap with the same name installed (and perhaps even active!).